doc/introduction/interfaces.rst

```diff
@@ -3,17 +3,104 @@
 
 .. _intro_interfaces:
 
-Interfaces and training
-=======================
+Gradients and training
+======================
 
 PennyLane offers seamless integration between classical and quantum computations. Code up quantum
-circuits in PennyLane, and connect them easily to the top scientific computing and machine learning
-libraries.
+circuits in PennyLane, compute :doc:`gradients of quantum circuits <glossary/quantum_gradient>`, and
+connect them easily to the top scientific computing and machine learning libraries.
+
+Gradients
+---------
+
+When creating a QNode, you can specify the :doc:`differentiation method
+<glossary/quantum_differentiable_programming>` that PennyLane should use whenever the gradient of
+that QNode is requested.
+
+.. code-block:: python
+
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.probs(wires=0)
+
+PennyLane currently provides the following differentiation methods for QNodes:
+
+Simulation-based differentiation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following methods use `reverse accumulation
+<https://en.wikipedia.org/wiki/Automatic_differentiation#Reverse_accumulation>`__ to compute
+gradients; a well-known example of this approach is backpropagation. These methods are **not** hardware compatible; they are only supported on
+*statevector* simulator devices such as :class:`default.qubit <~.DefaultQubit>`.
+
+However, for rapid prototyping on simulators, these methods typically out-perform forward-mode
+accumulators such as the parameter-shift rule and finite-differences. For more details, see the
+:doc:`quantum backpropagation <demos/tutorial_backprop>` demonstration.
+
+* ``"backprop"``: Use standard backpropagation.
+
+  This differentiation method is only allowed on simulator
+  devices that are classically end-to-end differentiable, for example
+  :class:`default.qubit <~.DefaultQubit>`. This method does *not* work on devices
+  that estimate measurement statistics using a finite number of shots; please use
+  the ``parameter-shift`` rule instead.
+
+* ``"adjoint"``: A form of backpropagation that takes advantage of the unitary or reversible
+  nature of quantum computation.
+
+  The `adjoint method <https://arxiv.org/abs/2009.02823>`__  reverses through the circuit after a
+  forward pass by iteratively applying the inverse (adjoint) gate. This method is similar to
+  ``"backprop"``, but trades off increased runtime with significantly lower memory usage.
```

Comment:
Is this true? i.e., "trades off increased runtime with significantly lower memory usage"?
We saw on benchmarks that adjoint was faster, and I'm not sure if there's a complexity argument to say backprop is faster?

Note that this sentence seems closer to the one we have for reversible:
"The reversible approach is similar to backpropagation, but trades off extra computation for enhanced memory efficiency."

This is what we say for adjoint:
"This method is similar to the reversible method, but has a lower time overhead and a similar memory overhead."

Overall, I need to think about it, but I'm not sure if there is an occasion where reversible is preferable.

Reply:
@trbromley this is a good point. Do you have any suggestions we should change it to?

Reply:
:thinking: Could we steer clear of making comparisons between the methods? I feel like probably only one option should be on offer eventually, with the others potentially available but not advertised. We could e.g., do a tutorial comparing?

But in the meantime, yes maybe we don't compare, or:
"This method is similar to ``"backprop"``, but has significantly lower memory usage and a similar runtime."

Actually, what is the runtime complexity of backprop?

Reply:
```suggestion
  ``"backprop"``, but has significantly lower memory usage and a similar runtime.
```

> Actually, what is the runtime complexity of backprop?

I am not 100% sure. I know it tends to have a constant overhead compared to the forward pass, with an additional memory requirement during the forward pass due to storing intermediate computations

Reply:
Great, yes if it has a constant overhead relative to the forward pass then it is similar complexity to the adjoint method, which is effectively 2 forward passes (one forward, one in reverse) - although I would need to think more carefully before being totally confident about this statement :laughing:.

Reply:
This definitely makes sense then --- with the adjoint method, we are not storing the intermediate computations (thus, less memory), _but_ we need to 'rewind' a few operations (thus, similar/slightly longer runtime)